RELEASE_NOTES.md

```diff
@@ -4,34 +4,304 @@
 
 #### Chef/ResourceSetsInternalProperties
 
+The `ResourceSetsInternalProperties` cop detects resources that set internal state properties used by built in Chef Infra resources. These undocumented properties should not be set in a resource block and doing so will cause unexpected behavior when running Chef Infra Client.
+
+`Examples`
+
+Service resource setting the running property:
+```ruby
+service 'foo' do
+  running true
+  action [:start, :enable]
+end
+```
+
 #### Chef/ResourceSetsNameProperty
 
+The `ResourceSetsNameProperty` cop detects a resource block with the `name` property set. The `name` property is a special property that is derviced from the name of the resource block and should not be changed with the block. Changing the name within a resource block can cause issues with reporting and notifications. If you wish to give your resources a more friendly name consider setting using setting a `name_property` which is available in all built-in Chef Infra resources. The name_property for each resource can be found in the [resource reference documentation](https://docs.chef.io/resource_reference.html)
+
+`Examples`
+
+Service resource incorrectly setting the name property:
+```ruby
+service 'Start the important service' do
+  name 'foo'
+  action [:start, :enable]
+end
+```
+
+Service resource correctly setting the service_name name property:
+```ruby
+service 'Start the important service' do
+  service_name 'foo'
+  action [:start, :enable]
+end
+```
+
+`Enabled by default`: True
+
+`Autocorrects`: No
+
 #### Chef/ResourceWithNoneAction
 
+The `ResourceWithNoneAction` cop detects the use of the `:none` action in a resource. The `:none` action is a common typo for the built-in `:nothing` action in all resources.
+
+`Examples`
+
+Service resource with the incorrect :none action:
+```ruby
+service 'my_service' do
+  action [:none]
+end
+```
+
+`Enabled by default`: True
+
+`Autocorrects`: Yes
+
 #### Chef/ChocolateyPackageUninstallAction
 
+The `ChocolateyPackageUninstallAction` cop detects a `chocolatey_package` resource that uses the `:uninstall` action. The uninstall action has been replaced with the `:remove` action and will error in Chef Infra Client 14+
+
+`Examples`
+
+chocolatey_package incorrectly setting the :uninstall action:
+
+```ruby
+chocolatey_package 'nginx' do
+  action :uninstall
+end
+```
+
+chocolatey_package correctly setting the :remove action:
+
+```ruby
+chocolatey_package 'nginx' do
+  action :remove
+end
+```
+
+`Enabled by default`: True
+
+`Autocorrects`: Yes
+
 #### Chef/LaunchdDeprecatedHashProperty
 
+The `LaunchdDeprecatedHashProperty` cop detects the use of the deprecated `hash` property in the `launchd` resource. The hash property was renamed to `plist_hash` in Chef Infra Client 13 and support for the `hash` name was removed in `Chef Infra Client.
```

Comment:
What version is this supposed to be per the end of the last sentence?

Reply:
Oh yep missed that.